fvm/src/syscalls/env.rs

```diff
@@ -0,0 +1,23 @@
+use super::Context;
+use crate::kernel::{Kernel, Result};
+
+pub fn timestamp(_context: Context<'_, impl Kernel>) -> Result<u64> {
+    todo!()
+}
+
+pub fn blockhash(
+    _context: Context<'_, impl Kernel>,
+    _block: u32,
+    _ret_off: u32,
+    _ret_len: u32,
+) -> Result<u32> {
+    todo!()
+}
+
+pub fn gas_limit(_context: Context<'_, impl Kernel>) -> Result<u64> {
+    todo!()
+}
+
+pub fn gas_price(_context: Context<'_, impl Kernel>) -> Result<u64> {
+    todo!()
+}
```

Comment:
This is not correct. We already have the base_fee under `network::base_fee`. The EVM `GASPRICE` opcode returns the `effective_gas_price`, which is:

	effective_gas_price = priority_fee_per_gas + block.base_fee_per_gas

The piece we're missing is the gas premium (in the invocation context?), and the EVM runtime would fetch both the base fee and the gas premium to sum them and honour the GASPRICE behaviour.

Reply:
(also needs to be a token amount)

Reply:
so you want to remove this syscall entirely? I can add the gas premium in the invocation context.

Reply:
I can also move the gas limit to the invocation context as well, since they are intrinsically related.

Reply:
Ok, instead of passing them in the env context, I added gas premium as an argument to Send and fished the limit from the gas tracker.
I also added both of them to the invocation context struct.